From 14e3d20e35aed3c676e57b8bfa0208943fe69c43 Mon Sep 17 00:00:00 2001
From: Cedric Neveux <cedric.neveux@nxp.com>
Date: Thu, 16 Nov 2017 15:13:39 +0000
Subject: [PATCH 2950/5242] MLK-16959: crypto: caam Add CAAM page 0 definition
 in device tree

commit  543560697919ea559bce86431cc5df38eb9204ba from
https://source.codeaurora.org/external/imx/linux-imx.git

Add definition of CAAM page 0 in the device tree.
This page is only accessible by the CPU in secure world.
this is defined by the secure-status.

Signed-off-by: Cedric Neveux <cedric.neveux@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/crypto/fsl-sec4.txt        |   43 ++++++++++++++++++++
 arch/arm/boot/dts/imx6qdl.dtsi                     |    9 ++++
 arch/arm/boot/dts/imx6sx.dtsi                      |    9 ++++
 arch/arm/boot/dts/imx6ul.dtsi                      |    9 ++++
 arch/arm/boot/dts/imx7d.dtsi                       |    9 ++++
 5 files changed, 79 insertions(+)

diff --git a/Documentation/devicetree/bindings/crypto/fsl-sec4.txt b/Documentation/devicetree/bindings/crypto/fsl-sec4.txt
index 8a185f9..1fbd00a 100644
--- a/Documentation/devicetree/bindings/crypto/fsl-sec4.txt
+++ b/Documentation/devicetree/bindings/crypto/fsl-sec4.txt
@@ -160,6 +160,49 @@ iMX6UL does only require three clocks
 	};
 
 =====================================================================
+SEC 4 Page 0
+
+Description
+
+    Most of CAAM's configuration registers are accessible in block0
+    of CAAM's register space. These registers are intended to be
+    accessed by specially privileged software (e.g. boot software,
+    hypervisor, secure operating system).
+
+PROPERTIES
+
+   - compatible
+      Usage: required
+      Value type: <string>
+      Definition: Must include "fsl,sec-v4.0-ctrl".
+
+   - reg
+      Usage: required
+      Value type: <prop-encoded-array>
+      Definition: A standard property.  Specifies the physical
+          address and length of the SEC4 configuration registers.
+
+   - secure-status
+      Usage: required
+      Value type: <string>
+      Definition: Must include "okay".
+
+   - status
+      Usage: required
+      Value type: <string>
+      Definition: Must include "disabled".
+
+EXAMPLE
+	sec_ctrl: ctrl@0 {
+		/* CAAM Page 0 only accessible */
+		/*	by secure world */
+		compatible = "fsl,sec-v4.0-ctrl";
+		reg = <0x2100000 0x1000>;
+		secure-status = "okay";
+		status = "disabled";
+	};
+
+=====================================================================
 Job Ring (JR) Node
 
     Child of the crypto node defines data processing interface to SEC 4
diff --git a/arch/arm/boot/dts/imx6qdl.dtsi b/arch/arm/boot/dts/imx6qdl.dtsi
index 16d66bf..e75b3e5 100644
--- a/arch/arm/boot/dts/imx6qdl.dtsi
+++ b/arch/arm/boot/dts/imx6qdl.dtsi
@@ -1051,6 +1051,15 @@
 					 <&clks IMX6QDL_CLK_EIM_SLOW>;
 				clock-names = "mem", "aclk", "ipg", "emi_slow";
 
+				sec_ctrl: ctrl@0 {
+					/* CAAM Page 0 only accessible */
+					/*	by secure world */
+					compatible = "fsl,sec-v4.0-ctrl";
+					reg = <0x2100000 0x1000>;
+					secure-status = "okay";
+					status = "disabled";
+				};
+
 				sec_jr0: jr0@1000 {
 					compatible = "fsl,sec-v4.0-job-ring";
 					reg = <0x1000 0x1000>;
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 859ad67..b8d1b37 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -958,6 +958,15 @@
 					 <&clks IMX6SX_CLK_EIM_SLOW>;
 				clock-names = "mem", "aclk", "ipg", "emi_slow";
 
+				sec_ctrl: ctrl@0 {
+					/* CAAM Page 0 only accessible */
+					/*	by secure world */
+					compatible = "fsl,sec-v4.0-ctrl";
+					reg = <0x2100000 0x1000>;
+					secure-status = "okay";
+					status = "disabled";
+				};
+
 				sec_jr0: jr0@1000 {
 					compatible = "fsl,sec-v4.0-job-ring";
 					reg = <0x1000 0x1000>;
diff --git a/arch/arm/boot/dts/imx6ul.dtsi b/arch/arm/boot/dts/imx6ul.dtsi
index d87f449..cabd5a5 100644
--- a/arch/arm/boot/dts/imx6ul.dtsi
+++ b/arch/arm/boot/dts/imx6ul.dtsi
@@ -918,6 +918,15 @@
 					 <&clks IMX6UL_CLK_CAAM_MEM>;
 				clock-names = "ipg", "aclk", "mem";
 
+				sec_ctrl: ctrl@0 {
+					/* CAAM Page 0 only accessible */
+					/*	by secure world */
+					compatible = "fsl,sec-v4.0-ctrl";
+					reg = <0x2140000 0x1000>;
+					secure-status = "okay";
+					status = "disabled";
+				};
+
 				sec_jr0: jr0@1000 {
 					compatible = "fsl,sec-v4.0-job-ring";
 					reg = <0x1000 0x1000>;
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index de4f47e..ab83b46 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -388,6 +388,15 @@
 			<&clks IMX7D_AHB_CHANNEL_ROOT_CLK>;
 		clock-names = "ipg", "aclk";
 
+		sec_ctrl: ctrl@0 {
+			/* CAAM Page 0 only accessible */
+			/*	by secure world */
+			compatible = "fsl,sec-v4.0-ctrl";
+			reg = <0x30900000 0x1000>;
+			secure-status = "okay";
+			status = "disabled";
+		};
+
 		sec_jr0: jr0@1000 {
 			 compatible = "fsl,sec-v4.0-job-ring";
 			 reg = <0x1000 0x1000>;
-- 
1.7.9.5

